pkg/controller/activegate/version_label.go

```diff
@@ -0,0 +1,48 @@
+package activegate
+
+import (
+	"context"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/controller/dao"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controller/parser"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controller/version"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func (r *ReconcileActiveGate) setVersionLabel(pods []corev1.Pod) error {
+	for i := range pods {
+		pod := &pods[i]
+		for _, status := range pod.Status.ContainerStatuses {
+			if status.Image == "" {
+				// If Image is not present, skip
+				continue
+			}
+
+			imagePullSecret, err := dao.GetImagePullSecret(r.client, pod)
+			if err != nil {
+				// Something wrong with pull secret, exit function entirely
+				return err
+			}
+
+			dockerConfig, err := parser.NewDockerConfig(imagePullSecret)
+			// If an error is returned, try getting the image anyway
+
+			versionLabel, err2 := version.GetVersionLabel(status.Image, dockerConfig)
```

Comment:
Do you actually get something on dockerConfig is there was an error? Or just using nil there?

Reply:
On error, `dockerConfig` is `nil`